control-plane/pkg/reconciler/channel/v2/channelv2.go

```diff
@@ -0,0 +1,636 @@
+	logger := kafkalogging.CreateReconcileMethodLogger(ctx, channel)
+
+	channel.Status.Subscribers = make([]v1.SubscriberStatus, 0)
+	var globalErr error
+	currentCgs := make([]*internalscg.ConsumerGroup, 0)
+	for i := range channel.Spec.Subscribers {
+		s := &channel.Spec.Subscribers[i]
+		logger = logger.With(zap.Any("subscription", s))
+		cg, err := r.reconcileConsumerGroup(ctx, channel, s, topicName, bootstrapServers)
+		if err != nil {
+			logger.Error("error reconciling subscription. marking subscription as not ready", zap.Error(err))
+			channel.Status.Subscribers = append(channel.Status.Subscribers, v1.SubscriberStatus{
+				UID:                s.UID,
+				ObservedGeneration: s.Generation,
+				Ready:              corev1.ConditionFalse,
+				Message:            fmt.Sprint("Subscription not ready. ", err),
+			})
+			globalErr = multierr.Append(globalErr, err)
+		} else {
+			currentCgs = append(currentCgs, cg) //adding reconciled consumer group to map
+			if cg.IsReady() {
+				logger.Debug("marking subscription as ready")
+				channel.Status.Subscribers = append(channel.Status.Subscribers, v1.SubscriberStatus{
+					UID:                s.UID,
+					ObservedGeneration: s.Generation,
+					Ready:              corev1.ConditionTrue,
+				})
+			} else {
+				topLevelCondition := cg.GetConditionSet().Manage(cg.GetStatus()).GetTopLevelCondition()
+				if topLevelCondition == nil {
+					msg := fmt.Sprint("Subscription not ready. ", "Failed to reconcile consumer group. ", "Consumer group not ready")
+					channel.Status.Subscribers = append(channel.Status.Subscribers, v1.SubscriberStatus{
+						UID:                s.UID,
+						ObservedGeneration: s.Generation,
+						Ready:              corev1.ConditionUnknown,
+						Message:            msg,
+					})
+					globalErr = multierr.Append(globalErr, errors.New(msg))
+				} else {
+					msg := fmt.Sprint("Subscription not ready. ", topLevelCondition.Reason, topLevelCondition.Message)
+					channel.Status.Subscribers = append(channel.Status.Subscribers, v1.SubscriberStatus{
+						UID:                s.UID,
+						ObservedGeneration: s.Generation,
+						Ready:              corev1.ConditionFalse,
+						Message:            msg,
+					})
+					globalErr = multierr.Append(globalErr, errors.New(msg))
+				}
+			}
+		}
+	}
+
+	// Get all consumer groups associated with this Channel
+	selector := labels.SelectorFromSet(map[string]string{internalscg.KafkaChannelNameLabel: channel.Name})
+	channelCgs, err := r.ConsumerGroupLister.ConsumerGroups(channel.GetNamespace()).List(selector)
+	if err != nil {
+		globalErr = multierr.Append(globalErr, err)
+	}
+	for _, cg := range channelCgs {
+		found := false
+		for _, ccg := range currentCgs {
+			if ccg.Name == cg.Name {
+				found = true
+			}
+		}
+		if !found { //ConsumerGroup needs to be deleted since it isn't associated with an existing subscriber (subscriber may have been deleted)
```

Comment:
```suggestion
		if !found { // ConsumerGroup needs to be deleted since it isn't associated with an existing subscriber (subscriber may have been deleted)
```